### DISCUSSION TITLE
blueprint no HASP device available

### DISCUSSION THREAD

**Discussion by Stasky1968** [HASwitchPlate/HASPone#69] [Q&A] [answered]
I have installed the sw on d1 mini 
The d1 mini is connected to wifi and is connected to the mqtt brocker (i'm using Mosquitto integreted in HA)
I halso upgrade the sw and now is 1.01

![image](https://user-images.githubusercontent.com/52465806/114190329-789bb780-994b-11eb-8fb1-c834fa0bfc11.png)

HA version
core-2021.3.4
Operating System
5.12

I have the blueprint in HA but when i whant to create a automation on "HASP Device Select the HASP device" i dont have any available device. I think that there are somting wrong on mqtt auto discovery.
I have add manuali on binary sensor

- platform: mqtt
    name: display_camera Connected
    device_class: connectivity
    state_topic: "hasp/display_camera/status"
    payload_on: "ON"
    payload_off: "OFF"
    availability_topic: "hasp/display_camera/status"
    payload_available: "ON"
    payload_not_available: "OFF"

the status of sensor is ON but this sensor is not see by the blueprint

sorry if i was not very clear but i just started with HA

**Comment by aderusha** [comment]
If you go to `Configuration` > `Devices` in Home Assistant, do you see your device `display_camera` listed there?  One quick fix that often works is to power cycle the HASPone which will re-send all of the discovery messages.

**Comment by Stasky1968** [comment]
Under Configuration > Devices i don't have display_camera , I have try to reboot the HASP but same. No display_camera on Devices.
There are oneway to add manualy ?

**Reply by aderusha** [reply]
No, this 100% relies on MQTT discovery.  If you go to Integrations > MQTT, select the 3 vertical dots, and select "System Options", is "Enable newly added entities" enabled?

**Reply by Stasky1968** [reply]
yes is enabled 

![image](https://user-images.githubusercontent.com/52465806/114193416-c8c84900-994e-11eb-9fdd-4320fd507161.png)

But i think that is something related to the last version of Mosquitto broker Current version: 5.1.1

the broker configuration is :

![image](https://user-images.githubusercontent.com/52465806/114194231-8f440d80-994f-11eb-811f-c1d0702fe67e.png)

**Reply by aderusha** [reply]
Let's try this: go to `Integrations` > `MQTT` > `Configure`, then enter `homeassistant/#` under `Listen to a topic`, then press `START LISTENING`.  Do you see a bunch of JSON objects displayed on the screen in response?

**Reply by Stasky1968** [reply]
Messaggio 5 ricevuto su homeassistant/number/display_camera/config alle 16:33:
{
    "name": "display_camera active page",
    "command_topic": "hasp/display_camera/command/page",
    "state_topic": "hasp/display_camera/state/page",
    "retain": true,
    "optimistic": true,
    "icon": "mdi:page-next-outline",
    "unique_id": "display_camera-68c63af8cd5f-page",
    "device": {
        "identifiers": [
            "display_camera-68c63af8cd5f"
        ],
        "name": "display_camera",
        "manufacturer": "HASwitchPlate",
        "model": "HASPone v1.0.0",
        "sw_version": 1.01
    }
}
QoS: 0 - Retain: true
Messaggio 4 ricevuto su homeassistant/number/plate01/config alle 16:33:
{
    "name": "plate01 active page",
    "command_topic": "hasp/plate01/command/page",
    "state_topic": "hasp/plate01/state/page",
    "retain": true,
    "optimistic": true,
    "icon": "mdi:page-next-outline",
    "unique_id": "plate01-68c63af8cd5f-page",
    "device": {
        "identifiers": [
            "plate01-68c63af8cd5f"
        ],
        "name": "plate01",
        "manufacturer": "HASwitchPlate",
        "model": "HASPone v1.0.0",
        "sw_version": 1.01
    }
}
QoS: 0 - Retain: true
Messaggio 3 ricevuto su homeassistant/sensor/display_camera/config alle 16:33:
{
    "name": "display_camera sensor",
    "json_attributes_topic": "hasp/display_camera/sensor",
    "state_topic": "hasp/display_camera/status",
    "unique_id": "display_camera-68c63af8cd5f-sensor",
    "icon": "mdi:cellphone-text",
    "device": {
        "identifiers": [
            "display_camera-68c63af8cd5f"
        ],
        "name": "display_camera",
        "manufacturer": "HASwitchPlate",
        "model": "HASPone v1.0.0",
        "sw_version": 1.01
    }
}
QoS: 0 - Retain: true
Messaggio 2 ricevuto su homeassistant/sensor/plate01/config alle 16:33:
{
    "name": "plate01 sensor",
    "json_attributes_topic": "hasp/plate01/sensor",
    "state_topic": "hasp/plate01/status",
    "unique_id": "plate01-68c63af8cd5f-sensor",
    "icon": "mdi:cellphone-text",
    "device": {
        "identifiers": [
            "plate01-68c63af8cd5f"
        ],
        "name": "plate01",
        "manufacturer": "HASwitchPlate",
        "model": "HASPone v1.0.0",
        "sw_version": 1.01
    }
}
QoS: 0 - Retain: true
Messaggio 1 ricevuto su homeassistant/light/display_camera/config alle 16:33:
{
    "name": "display_camera backlight",
    "command_topic": "hasp/display_camera/light/switch",
    "state_topic": "hasp/display_camera/light/state",
    "brightness_state_topic": "hasp/display_camera/brightness/state",
    "brightness_command_topic": "hasp/display_camera/brightness/set",
    "availability_topic": "hasp/display_camera/status",
    "brightness_scale": 100,
    "unique_id": "display_camera-68c63af8cd5f-backlight",
    "payload_on": "ON",
    "payload_off": "OFF",
    "payload_available": "ON",
    "payload_not_available": "OFF",
    "device": {
        "identifiers": [
            "display_camera-68c63af8cd5f"
        ],
        "name": "display_camera",
        "manufacturer": "HASwitchPlate",
        "model": "HASPone v1.0.0",
        "sw_version": 1.01
    }
}
QoS: 0 - Retain: true
Messaggio 0 ricevuto su homeassistant/light/plate01/config alle 16:33:
{
    "name": "plate01 backlight",
    "command_topic": "hasp/plate01/light/switch",
    "state_topic": "hasp/plate01/light/state",
    "brightness_state_topic": "hasp/plate01/brightness/state",
    "brightness_command_topic": "hasp/plate01/brightness/set",
    "availability_topic": "hasp/plate01/status",
    "brightness_scale": 100,
    "unique_id": "plate01-68c63af8cd5f-backlight",
    "payload_on": "ON",
    "payload_off": "OFF",
    "payload_available": "ON",
    "payload_not_available": "OFF",
    "device": {
        "identifiers": [
            "plate01-68c63af8cd5f"
        ],
        "name": "plate01",
        "manufacturer": "HASwitchPlate",
        "model": "HASPone v1.0.0",
        "sw_version": 1.01
    }
}

**Reply by Stasky1968** [reply]
this is what i recive after START LISTENING

**Comment by Stasky1968** [comment, marked as answer]
A the end I have disintalled the mosquitto addon , commit all mqtt reference in configuration.
reinstall the mosquitto add on and now works thanks

**Reply by aderusha** [reply]
Great, nice work friend!
